libs/server-sdk/tests/evaluator_tests.cpp

```diff
@@ -246,3 +249,30 @@ TEST_F(EvaluatorTests, FlagWithExperimentTargetingMissingContext) {
     ASSERT_EQ(*detail, Value(false));
     ASSERT_EQ(detail.Reason(), EvaluationReason::Fallthrough(false));
 }
+
+TEST_F(EvaluatorTests, ThreadSafeEvaluation) {
```

Comment:
This type of test is what I've done in the past, and I'm not really sure of a better way to do it. 

Basically spin up a bunch of threads and see if ASAN is triggered.